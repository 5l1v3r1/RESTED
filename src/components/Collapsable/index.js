import React, { PropTypes } from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { Button, Collapse, Row, Col } from 'react-bootstrap';

import Fonticon from 'components/Fonticon';
import * as Actions from 'store/config/actions';
import { isOpen } from 'store/config/selectors';

import { StyledCollapsable } from './StyledComponents';

export function Collapsable({ id, title, open, children, toggleCollapse }) {
  return (
    <StyledCollapsable>
      <Button
        bsStyle="link"
        onClick={e => {
          e.preventDefault();
          toggleCollapse(id, open);
        }}
      >
        <h4>
          {title}
          <Fonticon
            icon="angle-right"
            className={classNames({
              'fa-rotate-90': open,
            })}
          />
        </h4>
      </Button>
      <Collapse in={open}>
        <Row>
          <Col xs={12}>
            {children}
          </Col>
        </Row>
      </Collapse>
    </StyledCollapsable>
  );
}

Collapsable.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  toggleCollapse: PropTypes.func.isRequired,
  open: PropTypes.bool,
};

const mapStateToProps = (state, props) => ({
  open: isOpen(state, props),
});

export default connect(mapStateToProps, Actions)(Collapsable);

