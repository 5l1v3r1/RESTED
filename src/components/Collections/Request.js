import React from 'react';
import { connect } from 'react-redux';
import { DragSource, DropTarget } from 'react-dnd';
import { ListGroup } from 'react-bootstrap';
import flow from 'lodash.flow';

import IconButton from 'components/IconButton';
import * as Actions from 'store/collections/actions';

import { StyledRequest, AsideButtons, MainContent } from './StyledComponents';
import * as Type from './dropTypes';

/**
 * Specifies the drag source contract.
 * Only `beginDrag` function is required.
 */
const requestSource = {
  canDrag() {
    // TODO Disallow drag if editing name
    return true;
  },

  beginDrag({ id, index, collectionIndex }) {
    return { id, index, collectionIndex };
  },
};

const requestTarget = {
  hover(props, monitor) {
    const dragIndex = monitor.getItem().index;
    const dragCollectionIndex = monitor.getItem().collectionIndex;
    const dragUUID = monitor.getItem().id;
    const hoverIndex = props.index;
    const hoverCollectionIndex = props.collectionIndex;
    const hoverUUID = props.id;

    // Abort further processing if dragged item is hovering over itself
    if (dragUUID === hoverUUID) {
      return;
    }

    // Note: we're mutating the monitor item here.
    // Generally it's better to avoid mutations,
    // but it's good here for the sake of performance
    // to avoid expensive index searches.
    /* eslint-disable no-param-reassign */
    monitor.getItem().index = hoverIndex;
    monitor.getItem().collectionIndex = hoverCollectionIndex;
    /* eslint-enable no-param-reassign */

    // Update redux orders
    props.reorderRequest({
      // Source
      collectionIndex: dragCollectionIndex,
      requestIndex: dragIndex,
    }, {
      // Target
      collectionIndex: hoverCollectionIndex,
      requestIndex: hoverIndex,
    });
  },
};

function Request(props) {
  const {
    connectDragSource,
    connectDropTarget,
    isDragging,
    id,
    collectionIndex,
    method,
    url,
    deleteRequest,
  } = props;

  return connectDragSource(connectDropTarget(
    <div> {/* Need a wrapper div for React DnD support */}
      <StyledRequest isDragging={isDragging}>
        <ListGroup componentClass="div">
          <div className="list-group-item">
            <AsideButtons>
              <IconButton
                tooltip="Edit"
                icon="cog"
              />
              <IconButton
                tooltip="Delete"
                icon="trash"
                onClick={() => deleteRequest(id, collectionIndex)}
              />
            </AsideButtons>

            <MainContent>
              <h4>{method}</h4>
              {url}
            </MainContent>
          </div>
        </ListGroup>
      </StyledRequest>
    </div>,
  ));
}

export default flow(
  DropTarget(Type.Request, requestTarget, connector => ({
    connectDropTarget: connector.dropTarget(),
  })),
  DragSource(Type.Request, requestSource, (connector, monitor) => ({
    connectDragSource: connector.dragSource(),
    isDragging: monitor.isDragging(),
  })),
  connect(null, Actions),
)(Request);

