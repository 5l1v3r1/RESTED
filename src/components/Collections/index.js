import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Panel } from 'react-bootstrap';

import * as Actions from 'store/collections/actions';
import CollectionList from './CollectionList';

function Titlebar({ addCollection }) {
  return (
    <span className="clearfix pull-right">
      <button onClick={addCollection}>
        Add new collection
      </button>
    </span>
  );
}

Titlebar.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

function Collections({ addCollection }) {
  return (
    <Panel header={<Titlebar addCollection={addCollection} />}>
      <CollectionList />
    </Panel>
  );
}

Collections.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

export default connect(null, Actions)(Collections);

