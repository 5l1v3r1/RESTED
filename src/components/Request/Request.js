import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { reduxForm, Field, Fields, FieldArray, propTypes } from 'redux-form';
import { Panel } from 'react-bootstrap';

import URLField from './URLField';
import MethodField from './MethodField';
import HeadersField from './HeadersField';
import BasicAuthField from './BasicAuthField';
import BodyField from './BodyField';

import * as Actions from '../../store/request/actions';
import { DEFAULT_REQUEST } from '../../constants/constants';

function Titlebar() {
  return (
    <h2>
      Request
    </h2>
  );
}

function Request({ placeholderUrl, handleSubmit, sendRequest }) {
  return (
    <Panel header={<Titlebar />}>
      <form onSubmit={handleSubmit(sendRequest)}>
        <Field
          name="url"
          component={URLField}
          placeholderUrl={placeholderUrl}
        />
        <Field
          name="method"
          component={MethodField}
        />
        <FieldArray
          name="headers"
          component={HeadersField}
        />
        <Fields
          names={['basicAuth.username', 'basicAuth.password']}
          component={BasicAuthField}
        />
        <BodyField />
      </form>
    </Panel>
  );
}

Request.propTypes = {
  placeholderUrl: PropTypes.string,
  ...propTypes,
};

const formOptions = {
  form: 'request',
};

const mapStateToProps = ({ request: { useFormData, placeholderUrl } }) => ({
  useFormData,
  placeholderUrl,
  initialValues: DEFAULT_REQUEST,
});

export { Request };

// TODO this
/* eslint-disable no-func-assign */
Request = reduxForm(formOptions)(Request);
Request = connect(mapStateToProps, Actions)(Request);
/* eslint-enable no-func-assign */
export default Request;

