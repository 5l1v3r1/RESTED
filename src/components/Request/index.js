import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { reduxForm, Field, Fields, FieldArray, propTypes } from 'redux-form';
import { Panel, Form } from 'react-bootstrap';
import flow from 'lodash.flow';

import * as requestActions from 'store/request/actions';
import requestValidation from 'utils/requestValidation';
import { DEFAULT_REQUEST } from 'constants/constants';

import Titlebar from './Titlebar';
import URLField from './URLField';
import MethodField from './MethodField';
import HeadersField from './HeadersField';
import BasicAuthField from './BasicAuthField';
import BodyField from './BodyField';

export const requestForm = 'request';

function Request({ placeholderUrl, handleSubmit, sendRequest }) {
  return (
    <Panel header={<Titlebar />}>
      <Form horizontal onSubmit={handleSubmit(sendRequest)}>
        <Field
          name="url"
          component={URLField}
          placeholderUrl={placeholderUrl}
        />
        <Field
          name="method"
          component={MethodField}
        />
        <FieldArray
          name="headers"
          component={HeadersField}
        />
        <Fields
          names={['basicAuth.username', 'basicAuth.password']}
          component={BasicAuthField}
        />
        <BodyField />
      </Form>
    </Panel>
  );
}

Request.propTypes = {
  placeholderUrl: PropTypes.string,
  ...propTypes,
};

const formOptions = {
  form: requestForm,
  validate: requestValidation,
};

const mapStateToProps = ({ request: { useFormData, placeholderUrl } }) => ({
  useFormData,
  placeholderUrl,
  initialValues: DEFAULT_REQUEST,
});

export { Request };

export default flow(
  reduxForm(formOptions),
  connect(mapStateToProps, requestActions),
)(Request);

