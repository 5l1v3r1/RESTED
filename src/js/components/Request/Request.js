import React from 'react';
import { Col } from 'react-bootstrap';

export default function Request() {
  return (
    <Col xs={12} sm={8}>
      <h2>
        Request
      </h2>
    </Col>
  );
}

