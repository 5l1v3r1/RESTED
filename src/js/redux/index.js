import { combineReducers } from 'redux';
import { reducer as form } from 'redux-form';

import request from './request/reducer';

export default combineReducers({
  request,
  form,
});

