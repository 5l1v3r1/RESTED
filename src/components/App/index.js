import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Row } from 'react-bootstrap';
import HTML5Backend from 'react-dnd-html5-backend';
import { DragDropContext } from 'react-dnd';
import flow from 'lodash.flow';

import Header from 'components/Header';
import LeftPanel from 'components/LeftPanel';
import Request from 'components/Request';
import Response from 'components/Response';
import Modal from 'components/Modal';
import updateTheme from 'utils/updateTheme';
import { getTheme, getCollectionsMinimized } from 'store/options/selectors';
import { fetchOptions } from 'store/options/actions';
import { fetchUrlVariables } from 'store/urlVariables/actions';
import { THEMES } from 'constants/constants';
import './GlobalStyles';

import { LeftCol, RightCol } from './StyledComponents';

/*
 * This must be a React.Component because DragDropContext
 * attaches a ref to the component, which as we know will
 * not work with a stateless functional component.
 */
class App extends React.Component {
  constructor(props) {
    super(props);
    props.fetchOptions();
    props.fetchUrlVariables();

    updateTheme(props.theme);
  }

  componentWillReceiveProps(newProps) {
    if (newProps.theme !== this.props.theme) {
      updateTheme(newProps.theme);
    }
  }

  render() {
    const { collectionsMinimized } = this.props;

    return (
      <div>
        <Header />
        <Row>
          <LeftCol
            xsHidden
            sm={collectionsMinimized ? null : 4}
            collapsed={collectionsMinimized}
          >
            <aside>
              <LeftPanel />
            </aside>
          </LeftCol>
          <RightCol
            xs={12}
            sm={collectionsMinimized ? 12 : 8}
          >
            <main>
              <Request />
              <Response />
            </main>
          </RightCol>
        </Row>
        <Modal />
      </div>
    );
  }
}

App.propTypes = {
  fetchUrlVariables: PropTypes.func.isRequired,
  fetchOptions: PropTypes.func.isRequired,
  theme: PropTypes.oneOf(THEMES).isRequired,
  collectionsMinimized: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  theme: getTheme(state),
  collectionsMinimized: getCollectionsMinimized(state),
});

export default flow(
  connect(mapStateToProps, { fetchOptions, fetchUrlVariables }),
  DragDropContext(HTML5Backend),
)(App);

