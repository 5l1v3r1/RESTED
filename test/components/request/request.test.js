import React from 'react';
import { combineReducers, createStore } from 'redux'
import { Provider } from 'react-redux'
import { reduxForm, reducer as form  } from 'redux-form';
import renderer from 'react-test-renderer';

/* eslint-disable import/no-unresolved */
import { Request } from 'components/Request';

jest.mock('react-dom');

const makeStore = (initial = {}) => createStore(
  combineReducers({ form }),
  { form: initial }
);

test('Request renders correctly', () => {
  const onSubmitFail = jest.fn();
  const store = makeStore({
    testForm: {}
  })

  const Decorated = reduxForm({
    form: 'testForm',
    onSubmitFail
  })(Request);

  const tree = renderer.create(
    <Provider store={store}>
      <Decorated />
    </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
  expect(onSubmitFail).not.toHaveBeenCalled();
});
