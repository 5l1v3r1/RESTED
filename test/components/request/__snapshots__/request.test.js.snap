exports[`test should render correctly 1`] = `
<div
  className="panel panel-default"
  id={undefined}>
  <div
    className="panel-heading">
    <span>
      Request
    </span>
  </div>
  <div
    className="panel-body">
    <form
      onSubmit={[Function]}>
      <div
        className="form-group">
        <div
          className="col-sm-2">
          <label
            className="control-label"
            htmlFor="url">
            URL
          </label>
        </div>
        <div
          className="col-sm-10">
          <input
            className="form-control"
            id="url"
            name="url"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder={undefined}
            type="text"
            value="" />
        </div>
      </div>
      <div
        className="form-group">
        <div
          className="col-sm-2">
          <label
            className="control-label"
            htmlFor="method">
            Method
          </label>
        </div>
        <div
          className="col-sm-8">
          <select
            className="form-control"
            id="method"
            name="method"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Method"
            type={undefined}
            value="GET">
            <option
              value="GET">
              GET
            </option>
            <option
              value="POST">
              POST
            </option>
            <option
              value="PUT">
              PUT
            </option>
            <option
              value="DELETE">
              DELETE
            </option>
            <option
              value="PATCH">
              PATCH
            </option>
            <option
              value="HEAD">
              HEAD
            </option>
            <option
              value="JSONP">
              JSONP
            </option>
            <option
              value="CUSTOM">
              CUSTOM
            </option>
          </select>
        </div>
        <div
          className="col-sm-2">
          <button
            className="btn btn-primary"
            disabled={false}
            type="submit">
            Send request
          </button>
        </div>
      </div>
      <div
        className="row">
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button">
          <h4>
            Headers
            <i
              className="fa fa-angle-right"
              role="presentation" />
          </h4>
        </button>
        <div
          aria-expanded={null}
          className="collapse col-xs-12">
          <div
            className="form-group">
            <div
              className="col-xs-4">
              <input
                className="form-control"
                id="header.headers[0]"
                name="headers[0].name"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Name"
                type="text"
                value="" />
            </div>
            <div
              className="col-xs-5">
              <input
                className="form-control"
                id="header.headers[0]"
                name="headers[0].value"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Value"
                type="text"
                value="" />
            </div>
            <div
              className="col-xs-1">
              <button
                className="btn btn-default"
                disabled={false}
                id="removeHeaderButton0"
                onClick={[Function]}
                type="button">
                <i
                  className="fa fa-trash"
                  role="presentation" />
                <span
                  className="sr-only">
                  Remove header
                </span>
              </button>
            </div>
          </div>
          <div
            className="col-xs-12">
            <button
              className="btn btn-default"
              disabled={false}
              id="addHeaderButton"
              onClick={[Function]}
              type="button">
              <i
                className="fa fa-plus"
                role="presentation" />
              Add header
            </button>
          </div>
        </div>
      </div>
      <div
        className="row">
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button">
          <h4>
            Basic auth
            <i
              className="fa fa-angle-right"
              role="presentation" />
          </h4>
        </button>
        <div
          aria-expanded={null}
          className="collapse col-xs-12">
          <div
            className="form-group">
            BasicAuth
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
